src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -1021,8 +1024,10 @@ class MergeTreeData : public IStorage, public WithMutableContext
     /// Engine-specific methods
     BrokenPartCallback broken_part_callback;
 
-    String log_name;
-    Poco::Logger * log;
+    /// log_name will change during table RENAME. Use atomic_shared_ptr to allow concurrent RW.
+    /// NOTE clang-14 doesn't have atomic_shared_ptr yet. Use std::atomic* operations for now.
+    std::shared_ptr<String> log_name;
```

Comment:
Maybe remove `log_name` and return `getStorageID().getNameForLogs()` from `getLogName()`?